### PR TITLE
Improve error handling in form in give devices

### PR DESCRIPTION
**Pull request by marekrozmus**

### CODE REVIEW

**Review by RedFox0x20 — APPROVED** [review]
Very nice changes, heavily improves UX.

**Comment by audreyfeldroy** [comment]
Uh oh, sorry about the merge conflict @marekrozmus. I can resolve it in the morning with fresh eyes, unless you get to that first.

File: components/form/InputFormControl.js [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
   const handleChange = (event) => onChange(event.target.value);
 
   return (
     <div className="form-control">
       {label && (
         <label className="label" htmlFor={id}>
-          <span className="label-text">{label}</span>
+          <span className="label-text">{`${label}${
+            required ? ' *' : ''
+          }`}</span>
```
Comment (audreyfeldroy) [inline components/form/InputFormControl.js:22]: There's a way to do this in CSS: https://stackoverflow.com/questions/11197671/use-css-to-automatically-add-required-field-asterisk-to-form-inputs

Earlier I was thinking that a CSS `:after` solution might be the way to go, but now I'm sort of on the fence. @marekrozmus which way do you think is cleaner code, and why? @theflucs I'd be curious to hear your thoughts too.

File: components/form/InputFormControl.js [hunk 2]
```diff
@@ -1,25 +1,43 @@
...
         </label>
       )}
       <input
-        className="input input-bordered input-sm"
+        className={`input input-bordered input-sm ${
+          error ? 'input-error' : ''
+        }`}
```
Comment (audreyfeldroy) [inline components/form/InputFormControl.js:28]: Yay, I like it!

File: components/form/InputFormControl.js [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
         onChange={handleChange}
         placeholder={placeholder || label}
         type={type || 'text'}
         value={value}
       />
+      {error && (
+        <label className="label">
+          <span className="label-text-alt text-red-500">{error}</span>
+        </label>
+      )}
```
Comment (audreyfeldroy) [inline components/form/InputFormControl.js:40]: I like your way of showing this if there's an error.

File: pages/give-devices.js [hunk 4]
```diff
@@ -21,77 +20,76 @@ const COUNTRIES = [
...
+  country: COUNTRIES[0].value,
+  streetAddress: '',
+  streetAddress2: '',
+  city: '',
+  state: '',
+  postalCode: '',
+  phone: '',
+  originalPurchasePrice: '',
+  description: '',
+};
```
Comment (audreyfeldroy) [inline pages/give-devices.js:35]: Ooh, good idea.

File: pages/give-devices.js [hunk 5]
```diff
@@ -21,77 +20,76 @@ const COUNTRIES = [
...
+  },
+  state: {
+    error: 'Please add your state or province.',
+    validator: isEmptyValidator,
+  },
+  postalCode: {
+    error: 'Please add your postal code.',
+    validator: isEmptyValidator,
+  },
+};
```
Comment (audreyfeldroy) [inline pages/give-devices.js:64]: I like how you combined the validations into this object. It's really nice and readable.

File: pages/give-devices.js [hunk 6]
```diff
@@ -21,77 +20,76 @@ const COUNTRIES = [
...
+  const [errors, setErrors] = React.useState({});
 
-  const hideModal = () => setModalOpen(false);
-
-  const showModal = (message) => {
-    setModalText(message);
-    setModalOpen(true);
-  };
+  const handleChange = (fieldName) => (value) =>
+    setFormData({ ...formData, [fieldName]: value });
```
Comment (audreyfeldroy) [inline pages/give-devices.js:72]: Whoa, currying and the spread operator, fun stuff! Check this out @RedFox0x20 @ozer619 😄

**Review by audreyfeldroy — COMMENTED** [review]
I reviewed roughly half of this and started to get sleepy. I'll pick up with the other half tomorrow. So far it's all good, thanks @marekrozmus!

**Review by audreyfeldroy — CHANGES_REQUESTED** [review]
Nice work @marekrozmus. The only comment to address is the question in InputFormControl.js `handleChange()`.

File: components/form/InputFormControl.js (hunk 1 above)
Comment (marekrozmus) [reply components/form/InputFormControl.js]: done - please check if you like the solution :)

**Review by audreyfeldroy — APPROVED** [review]

**Comment by audreyfeldroy** [comment]
Looks great @marekrozmus and thank you so much! 

And I like `clsx`, thanks for introducing us to it.

**Pull request merged**